pkg/util/timestamp_note.go

```diff
@@ -69,13 +64,37 @@ func (t TimestampNote) MarshalText() ([]byte, error) {
 	return []byte(t.String()), nil
 }
 
+// validateMessgeImprint ensures that the supplied string matches the following format:
+// [sha256:]<64 hexadecimal characters>
+// where [sha256:] is optional
+func validateMessgeImprint(v string) error {
+	var prefix, hash string
+
+	split := strings.SplitN(v, ":", 2)
+	switch len(split) {
+	case 1:
+		hash = split[0]
+	case 2:
+		prefix = split[0]
+		hash = split[1]
+	}
+
+	s := struct {
```

Comment:
this is fine, would it make sense to expose/export this struct somewhere else though?

Reply:
Yeah, might not be the best file name but done, and shared code with the same validation in the client